uvicorn/protocols/http/h11_impl.py

```diff
@@ -146,7 +145,7 @@ def connection_lost(self, exc):
                 # Premature client disconnect
                 pass
 
-        self.message_event.set()
+        self.cycle.message_event.set()
```

Comment:
Do we need…?

```suggestion
        if self.cycle is not None:
            self.cycle.message_event.set()
```

(I see we check this higher up in this method, and we do in the HttpTools implementation.)

Reply:
I remember effectively I added the `if self.cycle is not None:` in the http_tools implementation because one test failed without it,
the fact that tests pass without it in the h11 implementation is probably due to the fact (dont quote me on that) that we may only have that failing test on a default config which by default will pick the httptools implementation.

Reply:
I added your suggestion on both httptools and h11 @florimondmanca